src/lib/components/workspace/member-info-popup/MemberInfoPopup.svelte

```diff
@@ -93,27 +110,36 @@
           </div>
         </div>
         <div class="position">
-          {#if (userType === TeamRole.OWNER && user.role === TeamRole.MEMBER) || (userType === TeamRole.ADMIN && user.role === TeamRole.MEMBER)}
+          {#if (userType === TeamRole.TEAM_OWNER && (isWorkspaceMemberInfo ? teamRole === TeamRole.TEAM_MEMBER : user.role === TeamRole.TEAM_MEMBER)) || (userType === TeamRole.TEAM_ADMIN && isWorkspaceMemberInfo ? teamRole === TeamRole.TEAM_MEMBER : user.role === TeamRole.TEAM_MEMBER)}
             <MemberDropdown
-              id={user.id}
+              workspaceId={user.workspaceId}
+              id={user.id + "owner-team-acess"}
               data={getPermissionsData()}
-              method={user.role ? user.role : ""}
-              onclick={handleDropdown}
+              method={isWorkspaceMemberInfo ? teamRole : user.role}
+              onclick={isWorkspaceMemberInfo
+                ? handleDropDownWorkspaceLevel
+                : handleDropdown}
             />
```

Comment:
this code seems to be same for memberinfopopup and member workspace. if so extract to a new file and then import in both these files

Reply:
even the else part . just compare once

Reply:
I think doing this will make the flow more complicated as  memberInfo we are using in Tile.svelte( i.e which is used for Team Invite ) and WorkspaceUserInfo.svelte ( which is used for workspace flow) .we are already using MemberDropdown for reusabiltity . adding another layer will complicate things . let me know what you can think. can work around something